aiohttp/web_protocol.py

```diff
@@ -602,8 +602,6 @@ async def start(self) -> None:
                                 self._keepalive_handle = loop.call_at(
                                     now + keepalive_timeout, self._process_keepalive
                                 )
-                    else:
-                        break
```

Comment:
Good catch, I'm just trying to figure out if the transport.close() below should be run in the case of a BaseException that is uncaught..
@bdraco Do you think it's fine like this?

Reply:
I think this is a risk of the transport not getting closed on BaseException. I'm not sure what that exception would be though. 

maybe something like...
```diff
diff --git a/aiohttp/web_protocol.py b/aiohttp/web_protocol.py
index 3cefd5df..f2dcf5aa 100644
--- a/aiohttp/web_protocol.py
+++ b/aiohttp/web_protocol.py
@@ -589,6 +589,9 @@ class RequestHandler(BaseProtocol):
             except Exception as exc:
                 self.log_exception("Unhandled exception", exc_info=exc)
                 self.force_close()
+            except BaseException:
+                self.force_close()
+                raise
             finally:
                 if self.transport is None and resp is not None:
                     self.log_debug("Ignored premature client disconnection.")
```

Reply:
> I'm not sure what that exception would be though.

Well, in most cases, it'll probably be a KeyboardInterrupt, which may or may not lead to the program exiting..

> maybe something like...

Seems fine with me.

Reply:
Currently, the tests don't even complete, so the current change doesn't seem correct. Maybe with the proposed change it will work correctly..

Reply:
Tests are still not completing. @fazledyn-or Any interest in finishing this PR?

Reply:
> Tests are still not completing. @fazledyn-or Any interest in finishing this PR?

It's been long since I had worked on it, and yeah- I'll work toward finishing the PR. However, I don't understand the reason behind introducing an empty `raise`.

Reply:
I think in the original code if a BaseException happened, it could trigger the `break`, which then leads to closing the transport outside the exception handler. We shouldn't be suppressing a BaseException though, so to be safe we want to force close on a BaseException and then reraise it.

Looking over that code again, it feels like that block of logic should maybe not be in the finally block...